app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -0,0 +1,135 @@
+      when "COUNTER" then
+        min = @ems.metrics_connect.counters.get_data(metric[:id], starts: 0, limit: 1, order: 'ASC')
+        max = @ems.metrics_connect.counters.get_data(metric[:id], starts: 0, limit: 1, order: 'DESC')
+      when "AVAILABILITY" then
+        min = @ems.metrics_connect.avail.get_data(metric[:id], starts: 0, limit: 1, order: 'ASC')
+        max = @ems.metrics_connect.avail.get_data(metric[:id], starts: 0, limit: 1, order: 'DESC')
+      else raise MetricValidationError, "Validation error: unknown type #{metric_type}"
+      end
+      [min[0]['timestamp'], max[0]['timestamp']]
+    end
+
+    def collect_metrics(metric_id, metric_type, starts, ends, bucket_duration)
+      case metric_type
+      when "GAUGE" then gauges(metric_id, starts, ends, bucket_duration)
+      when "COUNTER" then counters(metric_id, starts, ends, bucket_duration)
+      when "AVAILABILITY" then availabilities(metric_id, starts, ends, bucket_duration)
+      else raise MetricValidationError, "Validation error: unknown type #{metric_type}"
+      end
+    end
+
+    def gauges(metric_id, starts, ends, bucket_duration)
+      sort_and_normalize(
+        @ems.metrics_connect.gauges.get_data(metric_id, :starts => starts, :ends => ends,
+                                             :bucketDuration => bucket_duration))
+    end
+
+    def counters(metric_id, starts, ends, bucket_duration)
+      compute_derivates(
+        sort_and_normalize(
+          @ems.metrics_connect.counters.get_data(metric_id, :starts => starts, :ends => ends,
+                                                 :bucketDuration => bucket_duration)))
+    end
+
+    def availabilities(metric_id, starts, ends, bucket_duration)
+      sort_and_normalize(
+        @ems.metrics_connect.avail.get_data(metric_id, :starts => starts, :ends => ends,
+                                            :bucketDuration => bucket_duration))
+    end
+
+    def sort_and_normalize(data)
+      # Sorting and removing last entry because always incomplete
+      # as it's still in progress.
+      norm_data = (data.sort_by { |x| x['start'] }).slice(0..-2)
+      norm_data.reject { |x| x.values.include?('NaN') }
+    end
+
+    def compute_derivates(counters)
+      counters.each_cons(2).map do |prv, n|
+        # Add min, median, max, percentile95th, etc. if needed
+        {
+          'start' => n['start'],
+          'end'   => n['end'],
+          'avg'   => n['avg'] - prv['avg']
+        }
+      end
+    end
+
+    def process_data(metric, data)
+      processed = Hash.new { |h, k| h[k] = {} }
+      data.each do |x|
+        timestamp = Time.at(x['start'] / 1.in_milliseconds).utc.to_i
+        processed[timestamp][metric[:name]] = if metric[:type] == 'AVAILABILITY'
+                                                x['uptimeRatio']
+                                              else
+                                                x['avg']
+                                              end
```

Comment:
This might be a little cleaner to use the ternary operator...

```ruby
processed[timestamp][metric[:name]] = (metric[:type] == 'AVAILABILITY' ? x['uptimeRatio'] : x['avg'])
```